lib/components/AnimatedRegion.js

```diff
@@ -8,40 +8,53 @@ if (__DEV__) {
     );
   }
 }
-// const __Animated = Object.getPrototypeOf(AnimatedWithChildren);
-// if (__Animated.name !== 'Animated') console.error('AnimatedRegion could not obtain Animated base class');
 
-var _uniqueId = 1;
+const configTypes = [
+  'latitude',
+  'longitude',
+  'latitudeDelta',
+  'longitudeDelta',
+];
+
+const defaultValues = {
+  // probably want to come up with better defaults
+  latitude: 0,
+  longitude: 0,
+  latitudeDelta: 0,
+  longitudeDelta: 0,
+};
+
+let _uniqueId = 1;
 
 export default class AnimatedMapRegion extends AnimatedWithChildren {
-  constructor(valueIn) {
+  constructor(valueIn = {}) {
     super();
-    var value = valueIn || {
-      // probably want to come up with better defaults
-      latitude: 0,
-      longitude: 0,
-      latitudeDelta: 0,
-      longitudeDelta: 0,
-    };
-    this.latitude =
-      value.latitude instanceof Animated.Value
-        ? value.latitude
-        : new Animated.Value(value.latitude);
-    this.longitude =
-      value.longitude instanceof Animated.Value
-        ? value.longitude
-        : new Animated.Value(value.longitude);
-    this.latitudeDelta =
-      value.latitudeDelta instanceof Animated.Value
-        ? value.latitudeDelta
-        : new Animated.Value(value.latitudeDelta);
-    this.longitudeDelta =
-      value.longitudeDelta instanceof Animated.Value
-        ? value.longitudeDelta
-        : new Animated.Value(value.longitudeDelta);
+    this.useNativeDriver = !!valueIn.useNativeDriver;
+    this.latitude = this.setInitialValue(
+      valueIn.latitude,
+      defaultValues.latitude
+    );
+    this.longitude = this.setInitialValue(
+      valueIn.longitude,
+      defaultValues.longitude
+    );
+    this.latitudeDelta = this.setInitialValue(
+      valueIn.latitudeDelta,
+      defaultValues.latitudeDelta
+    );
+    this.longitudeDelta = this.setInitialValue(
+      valueIn.longitudeDelta,
+      defaultValues.longitudeDelta
+    );
     this._regionListeners = {};
   }
 
+  setInitialValue(value, fallback) {
```

Comment:
I am probably a bit nit-picky, but this **function itself** does not really _set_ the initial values, right? I agree, that finding a good name is hard, but maybe something along the lines of `getAnimatedValue` describes a bit better, what it is actually doing? And maybe it can be extracted from the class, since it isn't really coupled to it in any meaningful way and is rather independent?